pandas/core/frame.py

```diff
@@ -646,7 +646,7 @@ def _constructor(self) -> Callable[..., DataFrame]:
         return DataFrame
 
     def _constructor_from_mgr(self, mgr, axes):
-        if self._constructor is DataFrame:
+        if type(self) is DataFrame:
             # we are pandas.DataFrame (or a subclass that doesn't override _constructor)
             return self._from_mgr(mgr, axes=axes)
```

Comment:
Alternative fix:
```suggestion
            return DataFrame._from_mgr(mgr, axes=axes)
```

Since `_from_mgr` is a class method. 

That would avoid passing a manager object to the DataFrame(..) constructor. 
Or another alternative (but that makes it a bit more complicated), is to keep the `if self._constructor is DataFrame:` check in the else block, like:

```
...
else:
    if self._constructor is DataFrame:
        return DataFrame._from_mgr(mgr)
    else:
        return self._constructor(mgr) 

```

Reply:
I like the first approach you suggest here better since it's a class method. But also, this is in some internals I'm not very familiar with so happy to go with whatever other approach.

The performance benefit of my current approach seems small, and okay to go without, right?